rpcs3/Emu/Cell/lv2/sys_process.cpp

```diff
@@ -188,22 +189,72 @@ error_code sys_process_get_id2(u32 object, vm::ptr<u32> buffer, u32 size, vm::pt
 	return process_get_id(object, buffer, size, set_size);
 }
 
-error_code process_is_spu_lock_line_reservation_address(u32 addr, u64 flags)
+CellError process_is_spu_lock_line_reservation_address(u32 addr, u64 flags)
 {
 	if (!flags || flags & ~(SYS_MEMORY_ACCESS_RIGHT_SPU_THR | SYS_MEMORY_ACCESS_RIGHT_RAW_SPU))
 	{
 		return CELL_EINVAL;
 	}
 
-	// TODO
-	return CELL_OK;
+	// TODO: respect sys_mmapper region's access rights
+	switch (addr >> 28)
+	{
+	case 0x0: // Main memory
+	case 0x1: // Main memory
+	case 0x2: // User 64k (sys_memory)
+	case 0xc: // RSX Local memory
+	case 0xe: // RawSPU MMIO
+		break;
+
+	case 0xf: // Private SPU MMIO
+	{
+		if (flags & SYS_MEMORY_ACCESS_RIGHT_RAW_SPU)
```

Comment:
What does it mean? You can put reservation in RawSPU area, or it's just how syscall works?

Reply:
You can't, but that's how it works.
You can't use reservations on RSX local memory nor RSX context as well but the syscall returns CELL_OK.